### DISCUSSION TITLE
clusrank sturctures of grouping, applicability

### DISCUSSION THREAD

**Discussion by Kr-S** [wenjie2wang/clusrank#2] [Q&A] [answered]
Hi!

First, I would like to thank the creators for the hard work you did to create this package! @wenjie2wang, thank you so much!

I have a somewhat awkward question regarding research design and applicability of the package for a test with paired observations, and a puzzling console response using the package.

We are comparing two devices on the same 27 subjects multiple times over the course of a month. 
So it is a repeated measures design (so perhaps not optimal for Wilcoxon anyway), patients as clusters, some unbalanced numbers of observations per patient, but at each time point one measurement from each device (so balanced observations per cluster). I don't think cluster size (number of repeated measurements performed) should play a big role in the outcome of difference between devices, so it is not informative. 

Which method is appropriate to choose here? When using "rgl" with paired=TRUE, we got a rather weird output, perhaps this is noteworthy to you. Also 4 observations were dropped, which did not happen with the "ds" setting (but as mentionedbefore, I don't think we have informative cluster size). Sadly I cannot share the data for a reproducible sample, but here is our console output (I apologize for the awkward data notation of the student who performed this):

![WhatsApp Image 2022-09-12 at 17 14 09](https://user-images.githubusercontent.com/64777012/189841337-a9557491-3114-4a48-8a58-98a588f9756a.jpeg)

Because the output returned things like "data: $ and $" I wondered if the package cannot handle the data structure somehow. I believe the student used dplyr and tibbles, if that is of help.
I looked at the sample data provided in the package documentation, and it seems the cluster-units are always receiving one intervention each there.
In our case each cluster-unit receives multiple measurements with both devices. This instance is described as "sub-unit grouping" I believe, in your paper "Wilcoxon Rank-Based Tests for Clustered Data with R Package clusrank", but I struggled/failed to find an applied example.

Thank you again,

S

**Comment by wenjie2wang** [comment, marked as answer]
@Kr-S Thanks for asking the question here.

To avoid the strange names in the output summary, could you try to specify the argument `data = data_new_old1` in `clusWilcox.test()`?  I would expect the same testing results.

**Reply by Kr-S** [reply]
Thank you for the prompt reply @wenjie2wang ! I will tell him to try. 
Is the method appropriate for this structure, where 2 devices were compared on each patient multiple times, but in some patients more paired measurements were performed than in others? I apologize that this is not a programming question, but it was not fully clear to me from the documentation.

**Reply by wenjie2wang** [reply]
> Is the method appropriate for this structure, where 2 devices were compared on each patient multiple times, but in some patients more paired measurements were performed than in others?

Yes, I think so from your description.  Section 3 of Jiang, et al. (2020) introduced the RGL method for paired observations with imbalanced cluster sizes.  Some reference code is given in Section 5.2.

**Reply by Kr-S** [reply]
And indeed, there it is. Thank you again!

**Reply by wenjie2wang** [reply]
You are welcome!

**Reply by Kr-S** [reply]
might the deleted obervations be due to rank ties?

**Reply by wenjie2wang** [reply]
Yes, ties are dropped.

https://github.com/wenjie2wang/clusrank/blob/cf4d6959bafbdd4b6003459eba23cf761c779466/R/clussignrank.R#L72-L78
